tiledb/sm/storage_manager/open_array.cc

```diff
@@ -69,6 +69,18 @@ ArraySchema* OpenArray::array_schema() const {
   return array_schema_;
 }
 
+std::unordered_map<std::string, tdb_shared_ptr<ArraySchema>>&
+OpenArray::array_schemas() {
+  return array_schemas_;
+}
+
+tdb_shared_ptr<ArraySchema> OpenArray::get_array_schema(
+    const std::string& schema_name) {
+  return (array_schemas_.find(schema_name) == array_schemas_.end()) ?
```

Comment:
There is a more efficient way for this. Store `array_schemas_.find(schema_name)` and don't use `array_schemas_[schema_name]`.